core/backend/src/IModelDb.ts

```diff
@@ -2533,6 +2534,20 @@ export class SnapshotDb extends IModelDb {
       snapshot.close();
       throw err;
     }
+
+    if (sasTokenExpiry) {
+      const expiresIn = Date.parse(sasTokenExpiry) - Date.now();
+
+      setTimeout(async () => {
+        const unlock = await RpcConfiguration.requestContext.requestMutex.lock();

```

Comment:
this doesn't seem necessary or warranted to me. Why do you need the "mutex"?

Reply:
This is still work in progress and I probably need to think about it more, but I wanted to avoid any requests being processed during the reattach. I wasn't sure if the daemon would be usable without interruptions while V2CheckpointManager.attach is being executed so I decided to play it safe. But I admit I didn't think it through very well, so I'll check if this is actually needed.

Reply:
yes, the deamon will continue to work successfully during the refresh unless the previous token expires. You should request the refresh long before that happens. None of this complication for the RPC layer should be necessary.